diff --git a/ckan/lib/helpers.py b/ckan/lib/helpers.py
index 3261a4131..19267b038 100644
--- a/ckan/lib/helpers.py
+++ b/ckan/lib/helpers.py
@@ -725,20 +725,21 @@ def _make_menu_item(menu_item, title, **kw):
     This function is called by wrapper functions.
     '''
     _menu_items = config['routes.named_routes']
-    if menu_item not in _menu_items:
-        raise Exception('menu item `%s` cannot be found' % menu_item)
-    item = copy.copy(_menu_items[menu_item])
-    item.update(kw)
-    active = _link_active(item)
-    needed = item.pop('needed')
-    for need in needed:
-        if need not in kw:
-            raise Exception('menu item `%s` need parameter `%s`'
-                            % (menu_item, need))
-    link = _link_to(title, menu_item, suppress_active_class=True, **item)
-    if active:
-        return literal('<li class="active">') + link + literal('</li>')
-    return literal('<li>') + link + literal('</li>')
+    if menu_item in _menu_items:
+        item = copy.copy(_menu_items[menu_item])
+        item.update(kw)
+        active = _link_active(item)
+        needed = item.pop('needed')
+        for need in needed:
+            if need not in kw:
+                raise Exception('menu item `%s` need parameter `%s`'
+                                % (menu_item, need))
+        link = _link_to(title, menu_item, suppress_active_class=True, **item)
+        if active:
+            return literal('<li class="active">') + link + literal('</li>')
+        return literal('<li>') + link + literal('</li>')
+    else:
+        return literal('')
 
 
 @core_helper
