diff --git a/ckan/controllers/home.py b/ckan/controllers/home.py
index 2a4388860..75e58831d 100644
--- a/ckan/controllers/home.py
+++ b/ckan/controllers/home.py
@@ -43,9 +43,16 @@ class HomeController(base.BaseController):
             # package search
             context = {'model': model, 'session': model.Session,
                        'user': c.user, 'auth_user_obj': c.userobj}
+
+            facets = h.facets();
+            facets.append("vocab_ILRI_vocregions")
+            facets.append("vocab_ILRI_voccountries")
+            facets.append("vocab_ILRI_vocspecies")
+            facets.append("vocab_ILRI_vocsubjects")
+            facets.append("vocab_ILRI_vocproduct")
             data_dict = {
                 'q': '*:*',
-                'facet.field': h.facets(),
+                'facet.field': facets,
                 'rows': 4,
                 'start': 0,
                 'sort': 'views_recent desc',
@@ -63,6 +70,11 @@ class HomeController(base.BaseController):
                 'tags': _('Tags'),
                 'res_format': _('Formats'),
                 'license': _('Licenses'),
+                'vocab_ILRI_vocregions': _('Regions'),
+                'vocab_ILRI_voccountries': _('Countries'),
+                'vocab_ILRI_vocspecies': _('Species'),
+                'vocab_ILRI_vocsubjects': _('Subjects'),
+                'vocab_ILRI_vocproduct': _('Products'),
             }
 
         except search.SearchError:
